src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/AssemblyBuilderImpl.cs

```diff
@@ -0,0 +1,135 @@
+                => new AssemblyBuilderImpl(name, coreAssembly, assemblyAttributes);
+
+        private static void WritePEImage(Stream peStream, MetadataBuilder metadataBuilder, BlobBuilder ilBuilder)
+        {
+            // Create executable with the managed metadata from the specified MetadataBuilder.
+            var peHeaderBuilder = new PEHeaderBuilder(
+                imageCharacteristics: Characteristics.Dll // Start off with a simple DLL
+                );
+
+            var peBuilder = new ManagedPEBuilder(
+                peHeaderBuilder,
+                new MetadataRootBuilder(metadataBuilder),
+                ilBuilder);
+
+            // Write executable into the specified stream.
+            var peBlob = new BlobBuilder();
+            peBuilder.Serialize(peBlob);
+            peBlob.WriteContentTo(peStream);
+        }
+
+        internal void Save(Stream stream)
+        {
+            ArgumentNullException.ThrowIfNull(stream);
+
+            if (_module == null)
+            {
+                throw new InvalidOperationException(SR.AModuleRequired);
+            }
+
+            if (_previouslySaved) // Cannot save an assembly multiple times. This is consistent with Save() in .Net Framework.
+            {
+                throw new InvalidOperationException(SR.CannotSaveMultipleTimes);
+            }
+
+            // Add assembly metadata
+            var metadata = new MetadataBuilder();
+
+            metadata.AddAssembly(
+               metadata.GetOrAddString(value: _assemblyName.Name!),
+               version: _assemblyName.Version ?? new Version(0, 0, 0, 0),
+               culture: _assemblyName.CultureName == null ? default : metadata.GetOrAddString(value: _assemblyName.CultureName),
+               publicKey: _assemblyName.GetPublicKey() is byte[] publicKey ? metadata.GetOrAddBlob(value: publicKey) : default,
+               flags: AddContentType((AssemblyFlags)_assemblyName.Flags, _assemblyName.ContentType),
+#pragma warning disable SYSLIB0037 // Type or member is obsolete
+               hashAlgorithm: (AssemblyHashAlgorithm)_assemblyName.HashAlgorithm
+#pragma warning restore SYSLIB0037
+               );
+
+            // Add module's metadata
+            _module.AppendMetadata(metadata);
+
+            var ilBuilder = new BlobBuilder();
+            WritePEImage(stream, metadata, ilBuilder);
+            _previouslySaved = true;
+        }
+
+        private static AssemblyFlags AddContentType(AssemblyFlags flags, AssemblyContentType contentType)
+            => (AssemblyFlags)((int)contentType << 9) | flags;
+
+        internal void Save(string assemblyFileName)
+        {
+            ArgumentNullException.ThrowIfNull(assemblyFileName);
+
+            using var peStream = new FileStream(assemblyFileName, FileMode.CreateNew, FileAccess.Write);
```

Comment:
```suggestion
            using var peStream = new FileStream(assemblyFileName, FileMode.Create, FileAccess.Write);
```
.NET Framework overwrites existing file if it is there.